sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -229,5 +238,82 @@ describe(
         currentMSFAA.id,
       );
     });
+
+    it("Should cancel a pending MSFAA record and save a cancellation notification when an MSFAA cancellation record is received in the response file.", async () => {
+      // Arrange
+      // Create a MSFAA record in pending state with the same MSFAA number as in the msfaa-full-time-receive-file-with-single-cancellation-record.dat
+      const student = await saveFakeStudent(db.dataSource);
+      const studentUserId = student.user.id;
+      const notificationMessageType = NotificationMessageType.MSFAACancellation;
+      const pendingMSFAARecord = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaInitialValues: {
+              msfaaNumber: FULL_TIME_SAMPLE_MSFAA_NUMBER,
+              offeringIntensity: OfferingIntensity.fullTime,
+            },
+          },
+        ),
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_FULL_TIME_RECEIVE_FILE_WITH_SINGLE_CANCELLATION_RECORD,
+      ]);
+      // Act
+      const processResult = await processor.processMSFAA(job);
+      const cancelledMSFAARecord = await db.msfaaNumber.findOne({
+        select: {
+          cancelledDate: true,
+          newIssuingProvince: true,
+        },
+        where: {
+          id: pendingMSFAARecord.id,
+        },
+      });
+      const notification = await db.notification.findOne({
+        select: {
+          id: true,
+          dateSent: true,
+          messagePayload: true,
+          notificationMessage: { templateId: true },
+          user: { email: true, firstName: true, lastName: true },
+        },
+        relations: { notificationMessage: true, user: true },
+        where: {
+          notificationMessage: {
+            id: notificationMessageType,
+          },
+          user: {
+            id: studentUserId,
+          },
+        },
+      });
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_FULL_TIME_RECEIVE_FILE_WITH_SINGLE_CANCELLATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 0.",
+            "Cancelled MSFAA records (type C): 1.",
+            "Record from line 1, updated as cancelled.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      expect(cancelledMSFAARecord.cancelledDate).toBe("2021-11-24");
+      expect(cancelledMSFAARecord.newIssuingProvince).toBe("ON");
+      expect(notification.dateSent).toBe(null);
+      expect(notification.messagePayload).toStrictEqual({
```

Comment:
Thanks for adding this verification.
@ann-aot @guru-aot @dheepak-aot I believe that we can check the dynamic payload saved also as much as possible as part of the E2Es.

Reply:
Yes, this is good